Explorer/Assets/DCL/SDKComponents/SceneUI/Components/UITransformComponent.cs

```diff
@@ -1,26 +1,52 @@
 ﻿using Arch.Core;
+using DCL.ECSComponents;
 using DCL.Optimization.Pools;
+using DCL.SDKComponents.SceneUI.Utils;
+using Google.Protobuf.Collections;
 using System;
 using System.Collections.Generic;
 using UnityEngine.Pool;
 using UnityEngine.UIElements;
 
 namespace DCL.SDKComponents.SceneUI.Components
 {
-    public struct UITransformComponent : IPoolableComponentProvider<VisualElement>
+    public class UITransformComponent : IPoolableComponentProvider<UITransformComponent>
     {
         public VisualElement Transform;
         public EntityReference Parent;
         public HashSet<EntityReference> Children;
         public bool IsHidden;
         public int RightOf;
 
-        VisualElement IPoolableComponentProvider<VisualElement>.PoolableComponent => Transform;
-        Type IPoolableComponentProvider<VisualElement>.PoolableComponentType => typeof(VisualElement);
+        UITransformComponent IPoolableComponentProvider<UITransformComponent>.PoolableComponent => this;
+        Type IPoolableComponentProvider<UITransformComponent>.PoolableComponentType => typeof(UITransformComponent);
+
+        internal EventCallback<PointerDownEvent> currentOnPointerDownCallback;
+        internal EventCallback<PointerUpEvent> currentOnPointerUpCallback;
+
+        public PointerEventType? PointerEventTriggered;
+        public RepeatedField<PBPointerEvents.Types.Entry> RegisteredPointerEvents { get; internal set; }
+
+        public void Initialize(string componentName, Entity entity, ref PBUiTransform sdkModel)
+        {
+            Transform ??= new VisualElement();
+            Transform.name = UiElementUtils.BuildElementName(componentName, entity);
+            Parent = EntityReference.Null;
+            Children = HashSetPool<EntityReference>.Get();
+            IsHidden = false;
+            RightOf = sdkModel.RightOf;
+
+            PointerEventTriggered = null;
+            this.RegisterPointerCallbacks(
```

Comment:
Subscribing for every transform independently on `PBPointerEvents` is redundant: you will create callbacks for each `UITransform` but it's only needed when `PBPointerEvents` are present.
I propose you to subscribe in a separate function. For example you can store a flag whether subscriptions were done and call  it from `RegisterPointerEvents`

Reply:
Good point! I've made the subscriptions be made ONLY for UITransforms that also have the `PBPointerEvents` component 👍 Thanks for the suggestion!

Done here: https://github.com/decentraland/unity-explorer/pull/290/commits/975de9f738b3cf0530dbbe14ff5ec811f219d852